Model/InvoicePaymentMethod.php

```diff
@@ -0,0 +1,290 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Model;
+
+use OxidEsales\Eshop\Application\Model\Country;
+use OxidEsales\Eshop\Core\Exception\InputException;
+use OxidEsales\Eshop\Core\Field;
+use OxidEsales\Eshop\Core\Registry;
+
+use Wirecard\Oxid\Core\Helper;
+use Wirecard\Oxid\Core\SessionHelper;
+use Wirecard\Oxid\Extend\Model\Order;
+
+use Wirecard\PaymentSdk\Transaction\Transaction;
+
+/**
+ * Payment method implementation for Invoice payments
+ *
+ * @since 1.2.0
+ */
+abstract class InvoicePaymentMethod extends PaymentMethod
+{
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    public function getCheckoutFields()
+    {
+        $aCheckoutFields = [
+            'dateOfBirth' . self::getName() => [
+                'type' => $this->_getCheckoutFieldType(SessionHelper::isDateOfBirthSet(self::getName())),
+                'title' => Helper::translate('wd_birthdate_input'),
+                'description' => Helper::translate('wd_date_format_user_hint'),
+                'required' => true,
+            ],
+        ];
+
+        if ($this->_isPhoneMandatory()) {
+            $aCheckoutFields = array_merge($aCheckoutFields, [
+                'phone' . self::getName() => [
+                    'type' => $this->_getCheckoutFieldType(SessionHelper::isPhoneValid(self::getName())),
+                    'title' => Helper::translate('wd_phone'),
+                    'required' => true,
+                ],
+            ]);
+        }
+
+        if ($this->_checkSaveCheckoutFields($aCheckoutFields)) {
+            $aCheckoutFields = array_merge($aCheckoutFields, [
+                'saveCheckoutFields' . self::getName() => [
+                    'type' => 'select',
+                    'options' => [
+                        '1' => Helper::translate('wd_yes'),
+                        '0' => Helper::translate('wd_no'),
+                    ],
+                    'title' => Helper::translate('wd_save_to_user_account'),
+                ],
+            ]);
+        }
+
+        return $aCheckoutFields;
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @param Transaction $oTransaction
+     * @param Order       $oOrder
+     *
+     * @since 1.2.0
+     */
+    public function addMandatoryTransactionData(&$oTransaction, $oOrder)
```

Comment:
As payolution is extending this class and calling parent the basket and shipping is always added. Here you can find the mandatory fields for payolution maked as M https://doc.wirecard.com/payolution.html#payolution_Fields 

Basket and shipping is send if additional data is enabled.

Reply:
thanks, changed it